src/utils/compute_loss.jl

```diff
@@ -84,7 +85,26 @@ Returns a single loss value if `loss_spec` is provided, or a NamedTuple of losse
 """
 function _compute_loss end
 
-function assemble_loss(ŷ, y, y_nan, targets, loss_spec)
+# Wrapper for time-based subsetting - dispatches on array type for differentiability
+_select_time(ŷ_t::KeyedArray, time_keys) = ŷ_t(time = time_keys)  # KeyedArray: () syntax - view & differentiable
+_select_time(ŷ_t::AbstractDimArray, time_keys) = ŷ_t[time = At(time_keys)]  # DimArray: [] syntax - copy & differentiable
+
+function assemble_loss(ŷ, y::Union{KeyedArray{T, 3}, AbstractDimArray{T, 3}}, y_nan, targets, loss_spec) where {T}
```

Comment:
my original thinking for `assemble_loss` was to have only one, a generic function, and delegate/multidispatch on use cases to inner functions, such as `_get_target_y`, etc. We can come back to this later, once things work as expected.